tests/test_chiral_restraints.py

```diff
@@ -0,0 +1,506 @@
+
+    x0_inverted = [x0[0], x0[2], x0[1], x0[3], x0[4]]
+
+    inverted_restr_idxs = chiral_utils.setup_chiral_atom_restraints(mol, x0_inverted, 0)
+
+    # check the sign of the resulting idxs
+    k = 1000.0
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, normal_restr_idxs, k)) == 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0, inverted_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, normal_restr_idxs, k)) > 0)
+    assert np.all(np.asarray(U_chiral_atom_batch(x0_inverted, inverted_restr_idxs, k)) == 0)
+
+
+def test_setup_chiral_bond_restraints():
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 06082215002D
+
+  5  4  0  0  0  0            999 V2000
+   -0.5335   -0.0247    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
+    0.0285   -0.4092    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.0697   -1.2953    0.0000 Cl  0  0  0  0  0  0  0  0  0  0  0  0
+    0.6376   -0.0935    0.0000 N   0  0  0  0  0  0  0  0  0  0  0  0
+    1.2163   -0.3988    0.0000 F   0  0  0  0  0  0  0  0  0  0  0  0
+  2  4  2  0  0  0  0
+  4  5  1  0  0  0  0
+  1  2  1  0  0  0  0
+  2  3  1  0  0  0  0
+M  END
+$$$$
+""",
+        removeHs=False,
+    )
+
+    # needs to be batched in order for jax to play nicely
+    x0 = utils.get_romol_conf(mol)
+    src_atom = 1
+    dst_atom = 3
+    normal_restr_idxs, signs = chiral_utils.setup_chiral_bond_restraints(mol, x0, src_atom, dst_atom)
+
+    x0_inverted = [x0[0], x0[1], x0[2], x0[3], [0.07922, 0.06280, 0.0000]]
+
+    inverted_restr_idxs, inverted_signs = chiral_utils.setup_chiral_bond_restraints(
+        mol, x0_inverted, src_atom, dst_atom
+    )
+    k = 1000.0
+
+    assert np.all(np.asarray(U_chiral_bond_batch(x0, normal_restr_idxs, k, signs)) == 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0, inverted_restr_idxs, k, inverted_signs)) > 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_inverted, normal_restr_idxs, k, signs)) > 0)
+    assert np.all(np.asarray(U_chiral_bond_batch(x0_inverted, inverted_restr_idxs, k, inverted_signs)) == 0)
+
+
+def test_chiral_restraints_pyramidal():
+    mol = Chem.MolFromMolBlock(
+        """
+  Mrv2202 05192218063D
+
+  4  3  0  0  0  0            999 V2000
+   -0.0541    0.5427   -0.3433 N   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4368    0.0213    0.3859 H   0  0  0  0  0  0  0  0  0  0  0  0
+   -0.9636    0.0925   -0.4646 H   0  0  0  0  0  0  0  0  0  0  0  0
+    0.4652    0.3942   -1.2109 H   0  0  0  0  0  0  0  0  0  0  0  0
+  1  2  1  0  0  0  0
+  1  3  1  0  0  0  0
+  1  4  1  0  0  0  0
+M  END
+$$$$""",
+        removeHs=False,
+    )
+
+    ff = Forcefield.load_from_file("smirnoff_1_1_0_sc.py")
+    s_top = topology.BaseTopology(mol, ff)
+    x0 = utils.get_romol_conf(mol)
+
+    # check initial chirality
+    restr_idxs = np.array([0, 1, 2, 3])
+    vol = pyramidal_volume(*x0[restr_idxs])
+    assert vol > 0.5
+    system = s_top.setup_end_state(ff)
```

Comment:
I don't see `setup_end_state` defined?

Reply:
missed this in a bad cherry-pick from the single_topology PR,  just added the definitions now